src/BlockTemplatesController.php

```diff
@@ -25,27 +25,22 @@ class BlockTemplatesController {
 	private $template_parts_directory;
 
 	/**
-	 * Directory name of the block template directory.
+	 * Directory which contains all templates
 	 *
 	 * @var string
 	 */
-	const TEMPLATES_DIR_NAME = 'block-templates';
-
-	/**
-	 * Directory name of the block template parts directory.
-	 *
-	 * @var string
-	 */
-	const TEMPLATE_PARTS_DIR_NAME = 'block-template-parts';
+	const TEMPLATES_ROOT_DIR = 'templates';
 
 	/**
 	 * Constructor.
 	 */
 	public function __construct() {
 		// This feature is gated for WooCommerce versions 6.0.0 and above.
 		if ( defined( 'WC_VERSION' ) && version_compare( WC_VERSION, '6.0.0', '>=' ) ) {
-			$this->templates_directory      = plugin_dir_path( __DIR__ ) . 'templates/' . self::TEMPLATES_DIR_NAME;
-			$this->template_parts_directory = plugin_dir_path( __DIR__ ) . 'templates/' . self::TEMPLATE_PARTS_DIR_NAME;
+			$root_path = plugin_dir_path( __DIR__ ) . self::TEMPLATES_ROOT_DIR . DIRECTORY_SEPARATOR;
```

Comment:
Can we not replace `self::TEMPLATES_ROOT_DIR` with `BlockTemplateUtils::DIRECTORY_NAMES['TEMPLATES']` ?

Reply:
Technically, we can because they have the same value. However, the reason why I added `TEMPLATES_ROOT_DIR` in the first place is because they represent two different things:

1. `BlockTemplateUtils::DIRECTORY_NAMES['TEMPLATES']` represents the `block-templates` directory name in themes.
2. `BlockTemplatesController::TEMPLATES_ROOT_DIR` represents the root directory of all templates in our own repo. For example, we have email templates.

The just happen to be both named `templates` now, so we have the awkward path `templates/templates` (which makes me sad :'(). But technically they are separate entities that could also be changed independently at any point.

Does that make any sense?